calculator/src/main/java/com/market/banica/calculator/service/grpc/AuroraClientSideService.java

```diff
@@ -52,6 +56,29 @@
 
     }
 
+    public void announceInterests(List<String> products, String clientId) {
+
+        LOGGER.debug("Inside announceInterests method");
+        LOGGER.debug("Building blocking stub");
+        AuroraServiceGrpc.AuroraServiceBlockingStub blockingStub = getBlockingStub();
+
+        String message = String.join(",", products);
+
+        LOGGER.debug("Building request with parameters {}", message);
+        Aurora.AuroraRequest request = Aurora.AuroraRequest.newBuilder()
+                .setClientId(clientId)
+                .setTopic(MARKET_TOPIC_PREFIX + message)
+                .build();
+
+        Aurora.AuroraResponse auroraResponse = blockingStub.request(request);
+
+        if (auroraResponse.hasInterestsResponse()) {
+            return;
```

Comment:
maybe check if it doesn't have a response and then throw the exception this way the return here becomes obsolete - less code, easier to read

Reply:
Yes, forgot to invert the if, fixed.